src/js/index/todolist.js

```diff
@@ -0,0 +1,121 @@
+const $todoInput = document.querySelector("#new-todo-title");
+const $todoList = document.querySelector("#todo-list");
+const $todoCount = document.querySelector("strong");
+const $filters = document.querySelector(".filters");
+
+$todoInput.addEventListener("keyup", onAddTodoItem);
+$todoList.addEventListener("click", onToggleTodoItem);
+$todoList.addEventListener("dblclick", onEditTodoItem);
+$filters.addEventListener("click", onFilter);
+
+function onAddTodoItem(event) {
+    const todoTitle = event.target.value;
+    const todoList = document.getElementById("todo-list");
+    if (event.key === "Enter" && todoTitle !== "") {
+        todoList.insertAdjacentHTML("beforeend", renderTodoItemTemplate(todoTitle));
+        updateTodoCount();
+        event.target.value = "";
+        changeState();
+    }
+}
+
+function onToggleTodoItem(event) {
+    if (event.target.classList.contains("toggle")) {
+        event.target.closest("li").classList.toggle("completed");
+        event.target.closest("input").toggleAttribute("checked");
+        changeState()
+    } else if (event.target.classList.contains("destroy")) {
+        event.target.closest("li").remove();
+        updateTodoCount();
+        changeState();
+    }
+}
+
+function onEditTodoItem(event) {
+    if (event.target.classList.contains("label")) {
+        const originalValue = event.target.innerText;
+        event.target.closest("li").classList.add("editing");
+        event.target.parentElement.parentElement.querySelector(".edit").focus();
+        event.target.closest("li")
+            .addEventListener("keyup", function (event) {editTodoItem(event, originalValue)});
+    }
+}
+
+function editTodoItem(event, originalValue) {
+    switch (event.key) {
+        case "Enter" :
+            return updateTodoItem(event)
+        case "Escape" :
+            event.target.value = originalValue;
+            return event.target.closest("li").classList.remove("editing");
+    }
+}
+
+function updateTodoItem(event) {
+    const todos = document.querySelectorAll("#todo-list > li");
+    const changeValue = event.target.value;
+    todos.forEach((todo) => {
+        if (todo.id === event.target.closest("li").id) {
+            event.target.setAttribute("value", changeValue);
+            todo.querySelector(".label").innerText = changeValue;
+            todo.classList.remove("editing");
+        }
+    })
+}
+
+function renderTodoItemTemplate(title) {
+    return ` <li id="${String(Date.now())}">
```

Comment:
```suggestion
    return ` <li id="${Date.now()}">
```

JS에서는 형변환이 자동으로 이루어지므로, 반드시 String으로 변환시켜지 않아도 됩니다!